### PR TITLE
stage files from merge group properly

### LINKED ISSUE #5455: Cannot stage merged changes
### Description
During a rebase I had to merge changes. I cannot stage those changes using the SCM view (clicking the `+` button has no effect).

### Reproduction Steps
1. Rebase a branch so that changes need to be merged manually
2. Merge those changes
3. Try to stage those changes

**OS and Theia version:**
tested in gitpod.io

**Diagnostics:** 
there is no error on the console

### PR DESCRIPTION
**Pull request by akosyakov**
fix #5455

#### How to test

- Follow https://stackoverflow.com/questions/43702944/reproducing-git-merge-conflict-dd to create a repo with merge conflicts
- Check that stage and stage all commands are working from merge group

### CODE REVIEW

**Comment by akosyakov** [comment]
cc @32leaves

**Comment by kittaakos** [comment]
I could not spot any difference between the changes from here and the `master`; I ran into an error with both states: https://github.com/theia-ide/theia/issues/5463. Other than the error, when I staged and unstaged the files, both behaved the same.

**Comment by akosyakov** [comment]
You should be able to stage from merge conflicts. On master it does not work for individual files. It can only stage form changes.

File: packages/git/src/browser/git-scm-provider.ts [hunk 1]
```diff
@@ -260,17 +260,19 @@ export class GitScmProvider implements ScmProvider {
...
         } catch (error) {
             this.gitErrorHandler.handleError(error);
         }
     }
     async stage(uri: string): Promise<void> {
         try {
-            const { repository, unstagedChanges } = this;
-            if (unstagedChanges.some(change => change.uri === uri)) {
+            const { repository, stagedChanges } = this;
+            if (!stagedChanges.some(change => change.uri === uri)) {
```
Comment (AlexTugarev) [inline packages/git/src/browser/git-scm-provider.ts]: the way it was before should be good. this line breaks it. can you explain the intent, please?



I cannot stage this new change to a file which already was staged before.

Comment (AlexTugarev) [reply packages/git/src/browser/git-scm-provider.ts]: Maybe we should express the condition clearly; is this what you wanted to check for?
`const dirty = unstagedChanges.some(change => chan`ge.uri === uri) || mergeChanges.some(change => change.uri === uri);`

Comment (AlexTugarev) [reply packages/git/src/browser/git-scm-provider.ts]: just tested quickly; with the proposed change I could stage changes for already staged files and  it played well with the merge conflict changes.

**Comment by akosyakov** [comment]
@AlexTugarev @kittakos please feel free to open an alternative PR, i won't be around today

**Comment by AlexTugarev** [comment]
@kittaakos, I've update this PR. Please have a look.

**Comment by kittaakos** [comment]
> I cannot stage this new change to a file which already was staged before.

It worked. 👍

**Comment by akosyakov** [comment]
just had it again, pretty annoying that it is not possible to stage one file after another, merging it

**Pull request merged**